### ISSUE TITLE
Installation Issue

### ISSUE THREAD

**Issue by RunyuXia** [marcovarrone/cellcharter#69]
### Report

Hi! I’m encountering an issue while trying to install CellCharter. I followed the steps outlined in the README, but I keep running into an error. Here’s what I did: 

create -n cellcharter -c conda-forge python mamba
conda activate cellcharter
mamba install pytorch torchvision torchaudio pytorch-cuda=12.4 -c pytorch -c nvidia
pip install cellcharter

Thank you for your help!

<img width="909" alt="Image" src="https://github.com/user-attachments/assets/dfd6190a-91d7-44af-98d8-88f733f566cc" />

# packages in environment at /home/rx238/rds/hpc-work/miniconda3/envs/cellcharter:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
affine                    2.4.0                    pypi_0    pypi
aiobotocore               2.5.4                    pypi_0    pypi
aiohappyeyeballs          2.4.4                    pypi_0    pypi
aiohttp                   3.11.11                  pypi_0    pypi
aioitertools              0.12.0                   pypi_0    pypi
aiosignal                 1.3.2                    pypi_0    pypi
anndata                   0.10.9                   pypi_0    pypi
aom                       3.6.1                h59595ed_0    conda-forge
array-api-compat          1.10.0                   pypi_0    pypi
asciitree                 0.3.3                    pypi_0    pypi
async-timeout             5.0.1                    pypi_0    pypi
attrs                     24.3.0                   pypi_0    pypi
blas                      1.0                         mkl    conda-forge
botocore                  1.31.17                  pypi_0    pypi
brotli-python             1.1.0           py310hf71b8c6_2    conda-forge
bzip2                     1.0.8                h4bc722e_7    conda-forge
c-ares                    1.34.4               hb9d3cd8_0    conda-forge
ca-certificates           2024.12.14           hbcca054_0    conda-forge
cellcharter               0.3.2                    pypi_0    pypi
certifi                   2024.12.14         pyhd8ed1ab_0    conda-forge
cffi                      1.17.1          py310h8deb56e_0    conda-forge
charset-normalizer        3.4.1              pyhd8ed1ab_0    conda-forge
click                     8.1.8                    pypi_0    pypi
click-plugins             1.1.1                    pypi_0    pypi
cligj                     0.7.2                    pypi_0    pypi
cloudpickle               3.1.1                    pypi_0    pypi
colorcet                  3.1.0                    pypi_0    pypi
contourpy                 1.3.1                    pypi_0    pypi
cpp-expected              1.1.0                hf52228f_0    conda-forge
cpython                   3.10.16         py310hd8ed1ab_1    conda-forge
cuda-cudart               12.4.127                      0    nvidia
cuda-cupti                12.4.127                      0    nvidia
cuda-libraries            12.4.1                        0    nvidia
cuda-nvrtc                12.4.127                      0    nvidia
cuda-nvtx                 12.4.127                      0    nvidia
cuda-opencl               12.6.77                       0    nvidia
cuda-runtime              12.4.1                        0    nvidia
cuda-version              12.6                          3    nvidia
cycler                    0.12.1                   pypi_0    pypi
dask                      2025.1.0                 pypi_0    pypi
dask-image                2024.5.3                 pypi_0    pypi
datashader                0.16.3                   pypi_0    pypi
distributed               2025.1.0                 pypi_0    pypi
docrep                    0.3.2                    pypi_0    pypi
exceptiongroup            1.2.2                    pypi_0    pypi
fasteners                 0.19                     pypi_0    pypi
ffmpeg                    4.4.2           gpl_hdf48244_113    conda-forge
filelock                  3.16.1             pyhd8ed1ab_1    conda-forge
fmt                       11.0.2               h434a139_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_3    conda-forge
fontconfig                2.15.0               h7e30c49_1    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.55.3                   pypi_0    pypi
freetype                  2.12.1               h267a509_2    conda-forge
frozenlist                1.5.0                    pypi_0    pypi
fsspec                    2023.6.0                 pypi_0    pypi
geopandas                 1.0.1                    pypi_0    pypi
gettext                   0.22.5               he02047a_3    conda-forge
gettext-tools             0.22.5               he02047a_3    conda-forge
giflib                    5.2.2                hd590300_0    conda-forge
gmp                       6.3.0                hac33072_2    conda-forge
gmpy2                     2.1.5           py310he8512ff_3    conda-forge
gnutls                    3.7.9                hb077bed_0    conda-forge
h2                        4.1.0              pyhd8ed1ab_1    conda-forge
h5py                      3.12.1                   pypi_0    pypi
hpack                     4.0.0              pyhd8ed1ab_1    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_1    conda-forge
idna                      3.10               pyhd8ed1ab_1    conda-forge
igraph                    0.11.8                   pypi_0    pypi
imageio                   2.36.1                   pypi_0    pypi
importlib-metadata        8.5.0                    pypi_0    pypi
inflect                   7.5.0                    pypi_0    pypi
intel-openmp              2022.0.1          h06a4308_3633  
jinja2                    3.1.5              pyhd8ed1ab_0    conda-forge
jmespath                  1.0.1                    pypi_0    pypi
joblib                    1.4.2                    pypi_0    pypi
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.8                    pypi_0    pypi
krb5                      1.21.3               h659f571_0    conda-forge
lame                      3.100             h166bdaf_1003    conda-forge
lazy-loader               0.4                      pypi_0    pypi
lcms2                     2.16                 hb7c19ff_0    conda-forge
ld_impl_linux-64          2.43                 h712a8e2_2    conda-forge
legacy-api-wrap           1.4.1                    pypi_0    pypi
leidenalg                 0.10.2                   pypi_0    pypi
lerc                      4.0.0                h27087fc_0    conda-forge
libarchive                3.7.7                h4585015_3    conda-forge
libasprintf               0.22.5               he8f35ee_3    conda-forge
libasprintf-devel         0.22.5               he8f35ee_3    conda-forge
libblas                   3.9.0            16_linux64_mkl    conda-forge
libcblas                  3.9.0            16_linux64_mkl    conda-forge
libcublas                 12.4.5.8                      0    nvidia
libcufft                  11.2.1.3                      0    nvidia
libcufile                 1.11.1.6                      0    nvidia
libcurand                 10.3.7.77                     0    nvidia
libcurl                   8.11.1               h332b0f4_0    conda-forge
libcusolver               11.6.1.9                      0    nvidia
libcusparse               12.3.1.170                    0    nvidia
libdeflate                1.23                 h4ddbbb0_0    conda-forge
libdrm                    2.4.124              hb9d3cd8_0    conda-forge
libedit                   3.1.20240808    pl5321h7949ede_0    conda-forge
libegl                    1.7.0                ha4b6fd6_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libexpat                  2.6.4                h5888daf_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc                    14.2.0               h77fa898_1    conda-forge
libgcc-ng                 14.2.0               h69a702a_1    conda-forge
libgettextpo              0.22.5               he02047a_3    conda-forge
libgettextpo-devel        0.22.5               he02047a_3    conda-forge
libgl                     1.7.0                ha4b6fd6_2    conda-forge
libglvnd                  1.7.0                ha4b6fd6_2    conda-forge
libglx                    1.7.0                ha4b6fd6_2    conda-forge
libgomp                   14.2.0               h77fa898_1    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libidn2                   2.3.7                hd590300_0    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0            16_linux64_mkl    conda-forge
liblzma                   5.6.3                hb9d3cd8_1    conda-forge
libmamba                  2.0.5                h49b8a8d_1    conda-forge
libnghttp2                1.64.0               h161d5f1_0    conda-forge
libnpp                    12.2.5.30                     0    nvidia
libnsl                    2.0.1                hd590300_0    conda-forge
libnvfatbin               12.6.77                       0    nvidia
libnvjitlink              12.4.127                      0    nvidia
libnvjpeg                 12.3.1.117                    0    nvidia
libpciaccess              0.18                 hd590300_0    conda-forge
libpng                    1.6.45               h943b412_0    conda-forge
libsolv                   0.7.30               h3509ff9_0    conda-forge
libsqlite                 3.48.0               hee588c1_0    conda-forge
libssh2                   1.11.1               hf672d98_0    conda-forge
libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge
libstdcxx-ng              14.2.0               h4852527_1    conda-forge
libtasn1                  4.19.0               h166bdaf_0    conda-forge
libtiff                   4.7.0                hd9ff511_3    conda-forge
libunistring              0.9.10               h7f98852_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libva                     2.22.0               h8a09558_1    conda-forge
libvpx                    1.13.1               h59595ed_0    conda-forge
libwebp                   1.5.0                hae8dbeb_0    conda-forge
libwebp-base              1.5.0                h851e524_0    conda-forge
libxcb                    1.17.0               h8a09558_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxml2                   2.13.5               h0d44e9d_1    conda-forge
libzlib                   1.3.1                hb9d3cd8_2    conda-forge
lightning-utilities       0.11.9                   pypi_0    pypi
llvm-openmp               15.0.7               h0cdce71_0    conda-forge
llvmlite                  0.43.0                   pypi_0    pypi
locket                    1.0.0                    pypi_0    pypi
lz4-c                     1.10.0               h5888daf_1    conda-forge
lzo                       2.10              hd590300_1001    conda-forge
mamba                     2.0.5                h8871ed4_1    conda-forge
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                3.0.2           py310h89163eb_1    conda-forge
matplotlib                3.8.4                    pypi_0    pypi
matplotlib-scalebar       0.9.0                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
mkl                       2022.1.0           hc2b9512_224  
more-itertools            10.6.0                   pypi_0    pypi
mpc                       1.3.1                h24ddda3_1    conda-forge
mpfr                      4.2.1                h90cbb55_3    conda-forge
mpmath                    1.3.0              pyhd8ed1ab_1    conda-forge
msgpack                   1.1.0                    pypi_0    pypi
multidict                 6.1.0                    pypi_0    pypi
multipledispatch          1.0.0                    pypi_0    pypi
multiscale-spatial-image  2.0.2                    pypi_0    pypi
natsort                   8.4.0                    pypi_0    pypi
ncurses                   6.5                  h2d0b736_2    conda-forge
nettle                    3.9.1                h7ab15ed_0    conda-forge
networkx                  3.4.2              pyh267e887_2    conda-forge
nlohmann_json             3.11.3               he02047a_1    conda-forge
numba                     0.60.0                   pypi_0    pypi
numcodecs                 0.13.1                   pypi_0    pypi
numpy                     1.26.4                   pypi_0    pypi
ome-zarr                  0.10.3                   pypi_0    pypi
omnipath                  1.0.8                    pypi_0    pypi
openh264                  2.3.1                hcb278e6_2    conda-forge
openjpeg                  2.5.3                h5fbd93e_0    conda-forge
openssl                   3.4.0                h7b32b05_1    conda-forge
p11-kit                   0.24.1               hc5aa10d_0    conda-forge
packaging                 24.2                     pypi_0    pypi
pandas                    2.2.3                    pypi_0    pypi
param                     2.2.0                    pypi_0    pypi
partd                     1.4.2                    pypi_0    pypi
patsy                     1.0.1                    pypi_0    pypi
pillow                    11.1.0          py310h7e6dc6c_0    conda-forge
pims                      0.7                      pypi_0    pypi
pip                       24.3.1             pyh8b19718_2    conda-forge
platformdirs              4.3.6                    pypi_0    pypi
pooch                     1.8.2                    pypi_0    pypi
propcache                 0.2.1                    pypi_0    pypi
psutil                    6.1.1                    pypi_0    pypi
pthread-stubs             0.4               hb9d3cd8_1002    conda-forge
pyarrow                   19.0.0                   pypi_0    pypi
pycparser                 2.22               pyh29332c3_1    conda-forge
pyct                      0.5.0                    pypi_0    pypi
pygments                  2.19.1                   pypi_0    pypi
pynndescent               0.5.13                   pypi_0    pypi
pyogrio                   0.10.0                   pypi_0    pypi
pyparsing                 3.2.1                    pypi_0    pypi
pyproj                    3.7.0                    pypi_0    pypi
pysocks                   1.7.1              pyha55dd90_7    conda-forge
python                    3.10.16         he725a3c_1_cpython    conda-forge
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.10                    5_cp310    conda-forge
pytorch                   2.5.1           py3.10_cuda12.4_cudnn9.1.0_0    pytorch
pytorch-cuda              12.4                 hc786d27_7    pytorch
pytorch-lightning         2.5.0.post0              pypi_0    pypi
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2024.2                   pypi_0    pypi
pyyaml                    6.0.2           py310ha75aee5_1    conda-forge
rasterio                  1.4.3                    pypi_0    pypi
readline                  8.2                  h8228510_1    conda-forge
reproc                    14.2.5.post0         hb9d3cd8_0    conda-forge
reproc-cpp                14.2.5.post0         h5888daf_0    conda-forge
requests                  2.32.3             pyhd8ed1ab_1    conda-forge
rich                      13.9.4                   pypi_0    pypi
s3fs                      2023.6.0                 pypi_0    pypi
scanpy                    1.10.4                   pypi_0    pypi
scikit-image              0.25.0                   pypi_0    pypi
scikit-learn              1.6.1                    pypi_0    pypi
scipy                     1.15.1                   pypi_0    pypi
seaborn                   0.13.2                   pypi_0    pypi
session-info              1.0.0                    pypi_0    pypi
setuptools                75.8.0             pyhff2d567_0    conda-forge
shapely                   2.0.6                    pypi_0    pypi
simdjson                  3.11.6               h84d6215_0    conda-forge
six                       1.17.0                   pypi_0    pypi
sknw                      0.15                     pypi_0    pypi
slicerator                1.1.0                    pypi_0    pypi
sortedcontainers          2.4.0                    pypi_0    pypi
spatial-image             1.1.0                    pypi_0    pypi
spatialdata               0.2.6                    pypi_0    pypi
spatialdata-plot          0.2.8                    pypi_0    pypi
spdlog                    1.15.0               h10c9db5_0    conda-forge
squidpy                   1.6.1                    pypi_0    pypi
statsmodels               0.14.4                   pypi_0    pypi
stdlib-list               0.11.0                   pypi_0    pypi
svt-av1                   1.4.1                hcb278e6_0    conda-forge
sympy                     1.13.1                   pypi_0    pypi
tblib                     3.0.0                    pypi_0    pypi
texttable                 1.7.0                    pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
tifffile                  2025.1.10                pypi_0    pypi
tk                        8.6.13          noxft_h4845f30_101    conda-forge
toolz                     1.0.0                    pypi_0    pypi
torchaudio                2.5.1               py310_cu124    pytorch
torchgmm                  0.1.2                    pypi_0    pypi
torchmetrics              1.6.1                    pypi_0    pypi
torchtriton               3.1.0                     py310    pytorch
torchvision               0.20.1              py310_cu124    pytorch
tornado                   6.4.2                    pypi_0    pypi
tqdm                      4.67.1                   pypi_0    pypi
typeguard                 4.4.1                    pypi_0    pypi
typing_extensions         4.12.2             pyha770c72_1    conda-forge
tzdata                    2024.2                   pypi_0    pypi
umap-learn                0.5.7                    pypi_0    pypi
urllib3                   1.26.20                  pypi_0    pypi
validators                0.34.0                   pypi_0    pypi
wayland                   1.23.1               h3e06ad9_0    conda-forge
wayland-protocols         1.37                 hd8ed1ab_0    conda-forge
wheel                     0.45.1             pyhd8ed1ab_1    conda-forge
wrapt                     1.17.2                   pypi_0    pypi
x264                      1!164.3095           h166bdaf_2    conda-forge
x265                      3.5                  h924138e_3    conda-forge
xarray                    2024.11.0                pypi_0    pypi
xarray-dataclasses        1.9.1                    pypi_0    pypi
xarray-schema             0.0.3                    pypi_0    pypi
xarray-spatial            0.4.0                    pypi_0    pypi
xorg-libx11               1.8.10               h4f16b4b_1    conda-forge
xorg-libxau               1.0.12               hb9d3cd8_0    conda-forge
xorg-libxdmcp             1.1.5                hb9d3cd8_0    conda-forge
xorg-libxext              1.3.6                hb9d3cd8_0    conda-forge
xorg-libxfixes            6.0.1                hb9d3cd8_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
yaml-cpp                  0.8.0                h59595ed_0    conda-forge
yarl                      1.18.3                   pypi_0    pypi
zarr                      2.18.3                   pypi_0    pypi
zict                      3.0.0                    pypi_0    pypi
zipp                      3.21.0                   pypi_0    pypi
zstandard                 0.23.0          py310ha39cb0e_1    conda-forge
zstd                      1.5.6                ha6fb4c9_0    conda-forge

**Comment by marcovarrone** [comment]
Hi @RunyuXia, the error arises because the `spatialdata` package is using a feature of the `dask` package that has been removed in the last releases.

Using an older version of dask should fix the problem.

Try to run `pip install dask==2024.11.2` and let me know if it works!

**Comment by RunyuXia** [comment]
Thank you so much for your reply!

I encountered another error after downloading dask==2024.11.2, which seems to be an incompatibility issue with the spatialdata package.  I took the following steps to install the necessary packages:

pip install dask==2024.11.2
pip install dask-expr==1.1.19
pip install dask-image==2024.5.3
pip install squidpy==1.5.0
pip install distributed==2024.11.2
pip install spatial-image==1.1.0
pip install spatialdata==0.2.5.post0
pip install spatialdata-plot==0.2.7

**Comment by marcovarrone** [comment]
Thank you @RunyuXia! 

Actually, the problem may have been solved by the release of `spatialdata` v0.3.0 that came out few hours after your last message :)

I close the issue!
